### PR TITLE
Simplify _adv_getitem function

### PR DESCRIPTION
**Pull request by okuta**

### CODE REVIEW

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -2099,52 +2093,55 @@ cpdef ndarray _adv_getitem(ndarray a, slices):
...
+            else:
+                transp_b.append(i)
+                slices_b.append(s)
+        a = a.transpose(*(transp_a + transp_b))
+        slices = slices_a + slices_b
         li = 0
-        ri = p - 1
+        ri = len(transp_a) - 1
+
+    a_shepe = a.shape
```
Comment (wkentaro) [inline cupy/core/core.pyx]: typo? `a_shepe` -> `a_shape`

File: cupy/core/core.pyx [hunk 2]
```diff
@@ -2099,52 +2093,55 @@ cpdef ndarray _adv_getitem(ndarray a, slices):
...
     # build the strides
     strides = [1]
-    for i in range(ri, li, -1):
-        stride = a.shape[i] * strides[0]
-        strides.insert(0, stride)
+    for s in a.shape[ri:li:-1]:
+        strides.insert(0, s * strides[0])
 
     # convert all negative indices to wrap_indices
     for i in range(li, ri+1):
```
Comment (wkentaro) [inline cupy/core/core.pyx]: Should we use `six.moves.range` for `.pyx` file as well?

**Comment by wkentaro** [comment]
@yuyu2172 Do you have some comments?

File: cupy/core/core.pyx (hunk 2 above)
Comment (okuta) [inline cupy/core/core.pyx]: `range` is special function in Cython.
Please use `range`  in pyx file.
http://cython.readthedocs.io/en/latest/src/reference/language_basics.html#for-loops

File: cupy/core/core.pyx (hunk 1 above)
Comment (okuta) [inline cupy/core/core.pyx]: Yes, I fixed.

File: cupy/core/core.pyx (hunk 2 above)
Comment (wkentaro) [inline cupy/core/core.pyx]: That's new info for me! Thank you.

**Comment by yuyu2172** [comment]
Nope.

Is this going to be merged?
I am trying to make part of this function to work for both __getitem__ and __setitem__, and I would like to build on top of these commits.

**Comment by okuta** [comment]
Please proceed your PR without worrying about this PR @yuyu2172 .
I will rebase this PR.

If you can include this changes, you can include it.

**Comment by yuyu2172** [comment]
@okuta O.K.! Thanks!

**Comment by rezoo** [comment]
Any update on this?
If nobody disagree on this PR, I think it is better to merge it.

**Comment by rezoo** [comment]
LGTM

**Pull request merged**